.github/workflows/lint.yml

```diff
@@ -45,13 +45,15 @@ jobs:
           cache: pip
           cache-dependency-path: ./backend/pyproject.toml
           python-version: '3.12'
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
       - name: Install pre-commit
         run: |-
-          python -m pip install pre-commit
+          uv tool install pre-commit --with pre-commit-uv
```

Comment:
Theoretically, we don't actually have to install pre-commit, since we only use it once.
We could remove this and run `uvx --with pre-commit-uv pre-commit run --all-files` in the 'Run pre-commit' step